jdbc-v2/src/main/java/com/clickhouse/jdbc/ResultSetImpl.java

```diff
@@ -1221,50 +1153,38 @@ public Statement getStatement() throws SQLException {
 
     @Override
     public Object getObject(int columnIndex, Map<String, Class<?>> map) throws SQLException {
-        checkClosed();
-        return getObject(columnIndex);
+        return getObject(getSchema().columnIndexToName(columnIndex), map);
     }
 
     @Override
     public Ref getRef(int columnIndex) throws SQLException {
-        checkClosed();
-        if (!parentStatement.connection.config.isIgnoreUnsupportedRequests()) {
-            throw new SQLFeatureNotSupportedException("Ref is not supported.", ExceptionUtils.SQL_STATE_FEATURE_NOT_SUPPORTED);
-        }
-
-        return null;
+        return getRef(getSchema().columnIndexToName(columnIndex));
     }
 
     @Override
     public Blob getBlob(int columnIndex) throws SQLException {
-        checkClosed();
-        if (!parentStatement.connection.config.isIgnoreUnsupportedRequests()) {
-            throw new SQLFeatureNotSupportedException("Blob is not supported.", ExceptionUtils.SQL_STATE_FEATURE_NOT_SUPPORTED);
-        }
-
-        return null;
+        return getBlob(getSchema().columnIndexToName(columnIndex));
     }
 
     @Override
     public java.sql.Clob getClob(int columnIndex) throws SQLException {
-        checkClosed();
-        if (!parentStatement.connection.config.isIgnoreUnsupportedRequests()) {
-            throw new SQLFeatureNotSupportedException("Clob is not supported.", ExceptionUtils.SQL_STATE_FEATURE_NOT_SUPPORTED);
-        }
-
-        return null;
+        return getClob(getSchema().columnIndexToName(columnIndex));
     }
 
     @Override
     public java.sql.Array getArray(int columnIndex) throws SQLException {
-        checkClosed();
-        return getArray(reader.getSchema().columnIndexToName(columnIndex));
+        return getArray(getSchema().columnIndexToName(columnIndex));
     }
 
     @Override
     public Object getObject(String columnLabel, Map<String, Class<?>> map) throws SQLException {
         checkClosed();
-        return getObject(columnLabel);
+        log.debug("getObject(columnLabel={}, map={})", columnLabel, map);
+        if (map == null) {
```

Comment:
Just a note: 
In such cases I usually try to use one way of getting result. I would have a default map of type mappings, so only one generic method is used.